sklearn/base.py

```diff
@@ -292,6 +293,24 @@ def __repr__(self):
         return '%s(%s)' % (class_name, _pprint(self.get_params(deep=False),
                                                offset=len(class_name),),)
 
+    def __getstate__(self):
+        if type(self).__module__.startswith('sklearn.'):
+            return dict(self.__dict__.items(), __version__=__version__)
+        else:
+            return dict(self.__dict__.items())
```

Comment:
On 08/29/2016 03:54 PM, Gael Varoquaux wrote:
> Clarity and simplicity. Here we are relying on overriding the pickling 
> mechanism. We wouldn't need so. Less overriding of __ methods leads to 
> less tricky bug to debug.
I feel symmetry between ``__getstate__`` and ``__setstate__`` is more clear.
Adding an attribute that is only there for serialization purposes is 
somewhat non-obvious.
Our intent is to change the pickling behavior. I feel like overriding 
the pickling mechanism is the clearest way to do so.